Apps/Patient/src/Delegates/ClientRegistriesDelegate.cs

```diff
@@ -201,14 +201,14 @@ private void CheckResponseCode(string responseCode)
             {
                 // BCHCIM.GD.2.0018 Not found
                 this.logger.LogWarning("Client Registry did not find any records. Returned message code: {ResponseCode}", responseCode);
-                throw new ApiException(ErrorMessages.ClientRegistryRecordsNotFound, "ClientRegistriesDelegate.CheckResponseCode", HttpStatusCode.NotFound);
+                throw new ProblemDetailsException(ExceptionUtility.CreateProblemDetails(ErrorMessages.ClientRegistryRecordsNotFound, HttpStatusCode.NotFound, nameof(ClientRegistriesDelegate)));
```

Comment:
Keeping the ApiException makes more sense to me, then map it in the filter to problemdetail. This leaks problemdetail concept all the way down.

Reply:
Trying to keep it consistent as it was called that over at WaitingQueueWeb.  Maybe we need a discussion with @sslaws

Reply:
I think we should discuss; as Yossi says, I don't think there should be references to ProblemDetails inside any of our services or delegates.

Reply:
To be discussed in another story.